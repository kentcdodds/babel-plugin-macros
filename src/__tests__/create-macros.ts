import {createMacro} from '..'

test('throws error if it is not transpiled', () => {
  const untranspiledMacro = createMacro(() => {})
  expect(() =>
    untranspiledMacro({source: 'untranspiled.macro'}),
  ).toThrowErrorMatchingSnapshot()
})

test('attempting to create a macros with the configName of options throws an error', () => {
  expect(() =>
    createMacro(() => {}, {configName: 'options'}),
  ).toThrowErrorMatchingSnapshot()
})
