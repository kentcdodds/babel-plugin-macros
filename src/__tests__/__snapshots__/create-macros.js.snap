// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attempting to create a macros with the configName of options throws an error 1`] = `You cannot use the configName "options". It is reserved for babel-plugin-macros.`;

exports[`throws error if it is not transpiled 1`] = `The macro you imported from "untranspiled.macro" is being executed outside the context of compilation with babel-plugin-macros. This indicates that you don't have the babel plugin "babel-plugin-macros" configured correctly. Please see the documentation for how to configure babel-plugin-macros properly: https://github.com/kentcdodds/babel-plugin-macros/blob/master/other/docs/user.md`;
